### PR TITLE
[CARBONDATA-2710][Spark Integration] Refactor CarbonSparkSqlParser for better code reuse.

### PR DESCRIPTION
**Pull request by mohammadshahidkhan**
…

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [X] Any interfaces changed?
 NA
 - [X] Any backward compatibility impacted?
 NA
 - [X] Document update required?
NA
 - [X] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
      No change is core logic, only refactored the code to make it reusable.
 - [X] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. 
NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/6951/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5734/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5721/

**Comment by mohammadshahidkhan** [comment]
retest SDV please

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7007/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5759/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5785/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5762/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7054/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5831/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5798/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5868/

**Comment by mohammadshahidkhan** [comment]
retest this please

File: integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala [hunk 1]
```diff
@@ -169,220 +128,45 @@ class CarbonHelperSqlAstBuilder(conf: SQLConf,
...
-          "Table property 'streaming' should be either 'true' or 'false'")
-    }
+    val extraTableTuple = (cols, external, tableIdentifier, ifNotExists, colNames, tablePath,
+      tableProperties, properties, partitionByStructFields, partitionFields,
+      parser, sparkSession, selectQuery)
+    CarbonSparkSqlParserUtil
+      .createCarbonTable(createTableTuple, extraTableTuple)
   }
 
   private def validatePartitionFields(
```
Comment (KanakaKumar) [inline integration/spark2/src/main/scala/org/apache/spark/sql/parser/CarbonSparkSqlParser.scala]: Now this method has only 1 line of code. So, we can move this code line to caller and avoid this method completely.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7203/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/5986/

**Comment by KanakaKumar** [comment]
LGTM

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7216/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5875/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7247/

**Comment by mohammadshahidkhan** [comment]
Failled test case for the below build is unrelated. The same testcase is passing locally.
http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/7247/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/6022/

**Comment by mohammadshahidkhan** [comment]
retest SDV please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/5901/

**Pull request closed without merging**